### PR TITLE
Add better guard against location definition

### PR DESCRIPTION
**Pull request by acostalima**
Recently, I've run across a situation in a React Native environment where `location` was defined but `location.host` and `location.protocol` were not. The changes I propose herein defend the code better in such situation.

I've also took the opportunity to replace `self` with `globalThis` using [`globalthis`](https://github.com/es-shims/globalThis) shim.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,9 @@
         "node",
         "browser"
     ],
-    "dependencies": {},
+    "dependencies": {
+        "globalthis": "^1.0.1"
```
Comment (hugomrdias) [inline package.json:32]: why do we need globalthis dep ? `globalThis` or `self` doesnt exist in RN ?

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (acostalima) [reply package.json]: Not at all, my bad.
Yes, while `globalThis` does not exist, `self` does.

**Comment by hugomrdias** [comment]
can we add the RN test runner to the ci ?

**Comment by acostalima** [comment]
> can we add the RN test runner to the ci ?

I thought about this today as well, haha. Sure, I think it makes sense!
Shall we have a separate test suite for this?

**Comment by acostalima** [comment]
@hugomrdias can we merge this anyway to unblock https://github.com/ipfs/js-ipfs-utils/pull/91? I'll add RN test runner tomorrow in a separate PR.

**Pull request merged**